src/main/java/org/oasis_eu/portal/services/kernel/ApplicationInstanceStoreImpl.java

```diff
@@ -58,4 +62,19 @@
         logger.debug("Found {} pending instances", appInstancesList.size());
         return appInstancesList;
     }
+
+
+    public ApplicationInstance deletePendingInstance(String instanceId){
+        ResponseEntity<ApplicationInstance> respAppInstance = kernel.exchange(appsEndpoint + "/instance/{instance_id}",
+                HttpMethod.GET, null, ApplicationInstance.class, user(), instanceId);
+        String eTag = respAppInstance.getHeaders().getETag();
```

Comment:
Verify that the instance still is pending? (someone could have delete it already, or it might have been provisioned)

Ideally (but we won't rewrite the whole portal at that point, sigh), the etag should be (have been) kept along with the data up to the client, and sent down to the server when an action is made by the user, so that this GET request is not necessary. Only if the action failed with a 412 Precondition Failed then the portal could possibly retry the action (in this case, after doing a GET and checking that the status is still pending) and/or bubble the error up to the client, which can then refresh its data: the user will have the updated data and could either retry (if possible), or decide something else (for example, if the instance has been provisioned in between the first load of the data and the moment the user wanted to cancel the provisioning, he can then choose to keep the app, or delete it; it's up to him to decide what to do, depending on the reasons he tried to cancel the provisioning to begin with).